### ISSUE TITLE
Proposed UI changes

### ISSUE THREAD

**Issue by Harvie** [vlachoudis/bCNC#1057]
I have some proposals for UI change. It is based on my personal experience with bCNC.
And it also reflects the needs of people who are running bCNC on embedded devices. Like the guy on #1033 He will not be able to use bCNC if we'll not fix this before next release.

I've tried to implement all of these UI changes and it took me hours, but it always turned out that i don't know how tkinter works. @vlachoudis Maybe you could be a help with this one, since you are a specialist on bCNC UI :-)

Here is brief drawing of what i think should be done to improve usability:

![image](https://user-images.githubusercontent.com/276504/48817207-a0bbf900-ed45-11e8-9b4c-35d30036e4af.png)

**Comment by leofaoro** [comment]
I like your proposed changes, and I have a couple of comments.
- For the Config area, perhaps it’s more relevant to go into the File tab, as config/preferences typically go there (at least in Macs). Plus it is not a frequently used area once you get bCNC working.
- I think you should have the cycle start/pause/stop buttons always present regardless of what tab you’re in. At the very least the Stop button. As I have not installed physical buttons yet for my setup only the e-stop is setup), I have struggled more than once to change tabs to find the stop button, so having that easily reachable is good for safety.

Thanks
Leo

**Comment by Harvie** [comment]
>  cycle start/pause/stop buttons always present regardless of what tab you’re in

YES!

**Comment by Harvie** [comment]
Also i would like to have OPTION to show editor in CAM tab, since lots of times i need to use it to select the right block, when working with CAM features.

![image](https://user-images.githubusercontent.com/276504/48953399-0a0a5a80-ef46-11e8-9e96-261fef07487e.png)

**Comment by leofaoro** [comment]
> Maybe this button should go to "control" and rest to the "file".

My philosophy for this would be to have everything related to “setttings/options” bundled (as it is now). You probably don’t want to have to go to different places to change “global” settings.

**Comment by MARIOBASZ** [comment]
Hello, I just read this thread.
The button I use it sometimes, very punctually if I want to modify the height of Z surface or Z safe

**Comment by vlachoudis** [comment]
@Harvie the new update contains "almost" all of the changes. Moving the editor on the right is possible but requires more tweaking.
"Stock" is used by the default CAM operations (Cut, Profile, Drill...) when no information is provided on the material, thickness, safe z etc is taken from the stock

**Comment by Harvie** [comment]
Perfect! Thank you very much!

**Comment by Harvie** [comment]
Problem is that none of these buttons really work now:
![image](https://user-images.githubusercontent.com/276504/50871592-573e7080-13bc-11e9-9a89-b776818e4c3a.png)

in fact they do work, but they open dialog in CAM tab, which is not displayed at the moment you click it.
as temporary workaround i can suggest adding some call that would switch the tabs when clicked.

**Comment by vlachoudis** [comment]
@Harvie that why it was there in the first place  :)

**Comment by Harvie** [comment]
@vlachoudis makes sense :-D it would be cool if we were able to display and hide these panels independently on tab selected in ribbon. That would also enable us to view the g-code editor in CAM tab.

**Comment by MARIOBASZ** [comment]
Duplicating islands in editor is useful to me

**Comment by Harvie** [comment]
Maybe having the left panel with separate tabs would help. But i don't know. Probably it would be too confusing like this:

![image](https://user-images.githubusercontent.com/276504/53441863-0b956400-3a08-11e9-9ae9-6868f9a8383a.png)

But the tabs can be only virtual, so it would be possible to switch them based on what menu item people choose...

**Comment by Harvie** [comment]
Here is code to switch ribbon pages, i think it can be used:

```python
                # Restore last page
                self.pages["Probe"].tabChange() # Select "Probe:Probe" tab to show the dialogs!
                self.ribbon.changePage(Utils.getStr(Utils.__prg__,"page", "File"))

                probe = Page.frames["Probe:Probe"]
```
